tests/manage/test_expand_cluster_by_adding_backend_vol_rbd.py

```diff
@@ -0,0 +1,378 @@
+            instance_id=worker['id'],
+            name=f"{worker['name']}_extra_volume",
+            size=size,
+            timeout=150,
+            # device='/dev/sdf',
+            device=device_name,
+        )
+        time_out = 1
+        pod_obj = pod.get_ceph_tools_pod()
+        #  Wait for 10 mins in this loop to see if all osds including the newly added are in 'up' and 'in' state and
+        #  cluster is in healthy state
+        for i in range(60):
+            cmdoutput = pod_obj.exec_ceph_cmd('ceph -s')
+            up_osds = cmdoutput['osdmap']['osdmap']['num_up_osds']
+            in_osds = cmdoutput['osdmap']['osdmap']['num_in_osds']
+            pg_state = cmdoutput['pgmap']['pgs_by_state'][0]['state_name']
+            if up_osds == original_osd_count + osds_added + 1 and \
+                    in_osds == original_osd_count + osds_added + 1 and \
+                    pg_state == 'active+clean':
+                osds_added += 1
+                time_out = 0
+                logger.info("All osds including newly added are in 'up' and 'in' state and cluster is healthy")
+                break
+            else:
+                logger.info("Waiting for all osds including newly added to be in 'up' and 'in'. "
+                            "Waiting for 10 more seconds.....")
+                sleep(10)
+
+        if time_out:
+            raise TimeoutExpiredError(
+                "Timeout exceeded waiting for all the osds including the newly added to be 'up' and 'in' state"
+            )
+
+        logger.info("osd Addition status - Added : "+str(osds_added) + ". Requested: "+str(number_of_osds))
+        if osds_added == number_of_osds:
+            logger.info("Requested number of osds are now added to the system")
+            return
+
+
+def check_pod_restarts_count(before, after):
+    """
+    Compares the restart count of pods before and after adding backend volume. If restarted then returns True
+    Args:
+        before (dict): Dictionary of pods and restart count for each pod
+        after (dict): Dictionary of pods and restart count for each pod
+
+    Returns:
+        pod_restarted: boolean. True, if restart count changed. False, otherwise.
+    """
+    pod_restarted = False
+
+    for p in before:
+        if before[p] != after[p]:
+            pod_restarted = True
+            logging.error(f"The pod {p} has restarted. Restart count = {after[p]}")
+    return pod_restarted
+
+
+def check_osd_pods_added(before, after, expected_osd_pod_count):
+    """
+    Checks if osd pod corresponding to the newly added osd is shown in oc get pod output or not.
+
+    Args:
+        before (list): list of osd pods that were existing before the addition
+        after (list): list of  names after the osd was added
+        expected_osd_pod_count (int): number of expected osd pod count
+
+    Returns:
+        True, if the expected_osd_pod_count matches with the actual_osd_count observed.
```

Comment:
bool:  True if the expected_osd_pod_count matches with the actual_osd_count observed, False otherwise